Submitted By: Nico R. <n-roeser@gmx.net>
Date: 2004-08-17
Initial Package Version: 3.2.3
Upstream Status: In CVS
Origin: Upstream CVS
Description: This patch allows building kdegraphics with gcc-3.4.x.

$LastChangedBy$
$Date$

diff -Naur kdegraphics-3.2.3.orig/ksvg/ecma/ksvg_bridge.h kdegraphics-3.2.3+20040817T102208+0200/ksvg/ecma/ksvg_bridge.h
--- kdegraphics-3.2.3.orig/ksvg/ecma/ksvg_bridge.h	2003-10-26 10:52:36.000000000 +0000
+++ kdegraphics-3.2.3+20040817T102208+0200/ksvg/ecma/ksvg_bridge.h	2004-08-17 08:34:14.322264800 +0000
@@ -23,7 +23,9 @@
 
 #include <kdebug.h>
 
-#include "ksvg_lookup.h"
+#include <kjs/object.h>
+#include <kjs/lookup.h>
+#include <kjs/interpreter.h> // for ExecState
 
 namespace KJS
 {
@@ -90,7 +92,7 @@
 //			kdDebug(26004) << "KSVGBridge::put(), " << propertyName.qstring() << " Name: " << classInfo()->className << " Object: " << m_impl << endl;
 
 		// Try to see if we know this property (and need to take special action)
-		if(m_impl->put(exec, propertyName, value, attr))
+		if(this->m_impl->put(exec, propertyName, value, attr))
 			return;
 
 		// We don't -> set property in ObjectImp.
diff -Naur kdegraphics-3.2.3.orig/ksvg/ecma/ksvg_lookup.h kdegraphics-3.2.3+20040817T102208+0200/ksvg/ecma/ksvg_lookup.h
--- kdegraphics-3.2.3.orig/ksvg/ecma/ksvg_lookup.h	2003-08-17 11:49:23.000000000 +0000
+++ kdegraphics-3.2.3+20040817T102208+0200/ksvg/ecma/ksvg_lookup.h	2004-08-17 08:34:14.331261905 +0000
@@ -24,8 +24,7 @@
 #include <kjs/object.h>
 #include <kjs/lookup.h>
 #include <kjs/interpreter.h> // for ExecState
-
-class KSVGScriptInterpreter;
+#include "ksvg_scriptinterpreter.h"
 
 #define KSVG_GET_COMMON \
 public: \
diff -Naur kdegraphics-3.2.3.orig/ksvg/ecma/ksvg_scriptinterpreter.h kdegraphics-3.2.3+20040817T102208+0200/ksvg/ecma/ksvg_scriptinterpreter.h
--- kdegraphics-3.2.3.orig/ksvg/ecma/ksvg_scriptinterpreter.h	2003-05-10 09:32:50.000000000 +0000
+++ kdegraphics-3.2.3+20040817T102208+0200/ksvg/ecma/ksvg_scriptinterpreter.h	2004-08-17 08:34:14.343258045 +0000
@@ -100,7 +100,7 @@
 	{
 		ClassCtor* ctor = new ClassCtor(exec); // create the ClassCtor instance
 		KJS::Object newObject(new KSVGBridge<ClassCtor>(exec, ctor)); // create the bridge around it
-		exec->interpreter()->globalObject().put(exec, propertyName, newObject, Internal);
+		exec->interpreter()->globalObject().put(exec, propertyName, newObject, KJS::Internal);
 		return newObject;
 	}
 }
diff -Naur kdegraphics-3.2.3.orig/ksvg/impl/SVGHelperImpl.h kdegraphics-3.2.3+20040817T102208+0200/ksvg/impl/SVGHelperImpl.h
--- kdegraphics-3.2.3.orig/ksvg/impl/SVGHelperImpl.h	2003-08-27 20:08:25.000000000 +0000
+++ kdegraphics-3.2.3+20040817T102208+0200/ksvg/impl/SVGHelperImpl.h	2004-08-17 08:34:14.349256115 +0000
@@ -26,7 +26,7 @@
 #include "SVGLengthImpl.h"
 
 #include "ksvg_lookup.h"
-
+#include "SVGElementImpl.h"
 class QRect;
 class QString;
 
@@ -56,7 +56,7 @@
 		{
 			T *cast = dynamic_cast<T *>(element->ownerDoc()->getElementFromHandle(node.handle()));
 			if(cast)
-				cast->putValueProperty(element->ownerDoc()->ecmaEngine()->globalExec(), token, KJS::String(value), Internal);
+				cast->putValueProperty(element->ownerDoc()->ecmaEngine()->globalExec(), token, KJS::String(value), KJS::Internal);
 		}
 	}
 
diff -Naur kdegraphics-3.2.3.orig/ksvg/impl/SVGList.h kdegraphics-3.2.3+20040817T102208+0200/ksvg/impl/SVGList.h
--- kdegraphics-3.2.3.orig/ksvg/impl/SVGList.h	2004-01-17 12:52:40.000000000 +0000
+++ kdegraphics-3.2.3+20040817T102208+0200/ksvg/impl/SVGList.h	2004-08-17 08:34:14.353254828 +0000
@@ -48,7 +48,7 @@
 {
 public:
 	SVGList() { m_impl.setAutoDelete(false); }
-	SVGList(const SVGList &) { *this = other; }
+	SVGList(const SVGList &other) { *this = other; }
 	~SVGList() { clear(); }
 
 	SVGList<T> &operator=(const SVGList<T> &other)
